File: from-3101-to-3200/3194-minimum-average-of-smallest-and-largest-elements/solution.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        nums.sort()
        n = len(nums)
        l, r = 0, n - 1
        minAvg = float("inf")
        while l < r:
            avg = float(nums[l] + nums[r])/2.0
            if avg < minAvg:
                minAvg = avg
            l += 1
            r -= 1

        return minAvg

File: from-3101-to-3200/3194-minimum-average-of-smallest-and-largest-elements/solution.go
func minimumAverage(nums []int) float64 {
    n := len(nums)
    minAvg := math.MaxFloat64
    var avg float64
    sort.Ints(nums)
    l, r := 0, n - 1
    for l < r {
        avg = float64(nums[l] + nums[r])/2.0
        if avg < minAvg {
            minAvg = avg
        }
        l += 1
        r -= 1
    }
    return minAvg
}